src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,41 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance/party or player
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+
+    uint8 id = 0;
+
+    auto PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity);
+
+    if (PChar != nullptr)
+    {
+        if (PChar->PParty && PChar->PParty->m_PAlliance)
+        {
+            id = PChar->PParty->m_PAlliance->m_AllianceID;
+        }
+        else if (PChar->PParty)
+        {
+            id = PChar->PParty->GetPartyID();
+        }
+        else
+        {
+            id = PChar->id;
+        }
+    }
```

Comment:
If its never a intended to use this on anything but a player, you should prolly add a debug break
```cpp
DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
```
and/or maybe think about changing that luapush and return to only happen a result is attainable?
```cpp
    if (PChar)
    {
        if (PChar->PParty && PChar->PParty->m_PAlliance)
        {
            id = PChar->PParty->m_PAlliance->m_AllianceID;
        }
        else if (PChar->PParty)
        {
            id = PChar->PParty->GetPartyID();
        }
        else
        {
            id = PChar->id;
        }

        lua_pushnumber(L, id);
        return 1;
    }

    return 0;
}
```

Reply:
Just saw nfaletra's comments, was distracted by work and left my edit window up. I think pushing nil makes more sense.

Reply:
Yep I agree, just working through all the suggested changes while answering the door to kids as its halloween.